packages/wow-ui/src/hooks/useCheckedState.ts

```diff
@@ -13,17 +16,44 @@ interface CheckedStateProps {
 const useCheckedState = ({
   defaultChecked = false,
   checked: checkedProp,
-  disabled,
+  disabled: disabledProp,
+  value,
   onChange,
   onClick,
   onKeyDown,
 }: CheckedStateProps) => {
-  const [checked, setChecked] = useState<boolean>(() =>
-    checkedProp !== undefined ? checkedProp : defaultChecked
-  );
+  const group = useContext(MultiGroupContext);
+  const groupOnChange = group.onChange;
+  const groupChecked = group.checked?.includes(value);
+  const disabled = group.disabled || disabledProp || false;
 
+  const [checked, setChecked] = useState<boolean>(
+    groupChecked || checkedProp || defaultChecked
+  );
   const [pressed, setPressed] = useState<boolean>(false);
 
+  useEffect(() => {
+    if (groupChecked !== undefined) {
+      setChecked(groupChecked);
+    }
+  }, [groupChecked]);
+
+  useEffect(() => {
+    if (groupChecked === undefined && checkedProp !== undefined) {
+      setChecked(checkedProp);
+    }
+  }, [checkedProp, groupChecked]);
+
+  const toggleCheckedState = (value: string) => {
+    if (groupOnChange) {
+      if (!disabled) groupOnChange(value);
```

Comment:
p5;
조건문을 하나로 합쳐볼 수 있을 것 같아요!

```suggestion
 if (groupOnChange && !disabled) {
      groupOnChange(value);
    } 
```

Reply:
```
const toggleCheckedState = (value: string) => {
    if (disabled) return;

    if (groupOnChange) {
      groupOnChange(value);
    } else if (onChange) {
      onChange();
    } else {
      setChecked((prev) => !prev);
    }
  };
```
기존에는 disabled 상태를 다 고려하지 못한 거 같아서 일단 이렇게 변경했어요!
여기서 추가적으로 개선하면 좋겠다 싶으면 댓글 남겨주세요~!

Reply:
넵 전 좋습니다!